src/Shared/UrlDecoder/UrlDecoder.cs

```diff
@@ -345,5 +344,293 @@ private static bool SkipUnescape(int value1, int value2, bool isFormEncoding)
 
             return false;
         }
+
+        /// <summary>
+        /// Unescape a URL path
+        /// </summary>
+        /// <param name="source">The char span represents a UTF8 encoded, escaped url path.</param>
+        /// <param name="destination">The char span where unescaped url path is copied to.</param>
+        /// <returns>The length of the char sequence of the unescaped url path.</returns>
+        /// <remarks>
+        /// Form Encoding is not supported compared to the <see cref="DecodeRequestLine(ReadOnlySpan{byte}, Span{byte}, bool)" />
+        /// for performance gains, as current use-cases does not require it.
+        /// </remarks>
+        public static int DecodeRequestLine(ReadOnlySpan<char> source, Span<char> destination)
+        {
+            if (destination.Length < source.Length)
+            {
+                throw new ArgumentException(
+                    "Length of the destination char span is less then the source.",
+                    nameof(destination));
+            }
+
+            // This requires the destination span to be larger or equal to source span
+            source.CopyTo(destination);
+            return DecodeInPlace(destination.Slice(0, source.Length));
+        }
+
+        /// <summary>
+        /// Unescape a URL path in place.
+        /// </summary>
+        /// <param name="buffer">The char span represents a UTF8 encoded, escaped url path.</param>
+        /// <returns>The number of the chars representing the result.</returns>
+        /// <remarks>
+        /// The unescape is done in place, which means after decoding the result is the subset of
+        /// the input span.
+        /// Form Encoding is not supported compared to the <see cref="DecodeInPlace(Span{byte}, bool)" />
+        /// for performance gains, as current use-cases does not require it.
+        /// </remarks>
+        public static int DecodeInPlace(Span<char> buffer)
+        {
+            int position = buffer.IndexOf('%');
+            if (position == -1)
+            {
+                return buffer.Length;
+            }
+
+            // the slot to read the input
+            var sourceIndex = position;
```

Comment:
Compared to the byte implementation, this is a difference: using the first occurrence of % as the starting position.

Reply:
Included